gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/pdeps/ResolveProductDependenciesTask.java

```diff
@@ -135,16 +145,6 @@ public final void resolve() throws IOException {
             allProductDependencies.merge(productId, discoveredDependency, (declaredDependency, _newDependency) -> {
                 ProductDependency mergedDependency =
                         mergeDependencies(productId, declaredDependency, discoveredDependency);
-                if (mergedDependency.equals(discoveredDependency)) {
```

Comment:
I don't think this is quiet right, you still want this error to be produced. The problem is that you want to merge the discovered dependencies before attempting to merge them with the declared dependencies